Provide a basic style for .callout-box

ugultopu opened the pull request:
Currently there are no styles for `.callout-box` as discussed in #2158. This PR aims to provide some, possibly temporary, styling to the class that resembles its old style, so that it looks somewhat decent.

Feel free to replace the hardcoded values with variables with an edit.

# Before
<img src="https://i.ibb.co/LCjvwrC/Before.png" alt=".callout-box before changes">

# After
<img src="https://i.ibb.co/QJtp0Ph/After.png" alt=".callout-box after changes">

mattiapontonio commented:
Floating box of text... 🤔? Not sure.

ugultopu commented:
That's [how it was done before](https://web.archive.org/web/20201101052337/https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode), that's why I did it that way. As I said, the PR is so that there is at least _some_ style. Right now, `.callout-box` appears wrong. The PR can be edited as well.

mattiapontonio commented:
As You want but You can _also_ change the content https://github.com/mdn/content
and use something that is styled

IMHO

ugultopu commented:
Multiple articles to change vs. one style to add. Also I don't know what to use to replace them.

Be my guest for changing all articles that use the class.

mattiapontonio commented:
Maybe.
First contribute, a simple count 🔢
https://github.com/mdn/content/search?q=callout-box&type=

So lets organize...
Where keep tracks of **13** PR?
What Is the _prototype_ PR?

peterbe commented:
@ugultopu Great work! Ignore the previous comments. @schalkneethling will review this. The issue is well described and your PR is a solution that issue.

peterbe commented:
@ugultopu Did you see that you forgot to run Prettier on `client/src/document/index.scss`
You can probably just run `yarn prettier-format` in your branch and commit.

ugultopu commented:
@peterbe I did it using GitHub editor, didn't checkout the project. But you're right, I'm going to check it out and fix the errors.

ugultopu commented:
@peterbe I can't seem to build it locally. I give `content/files` as `CONTENT_ROOT` but the build fails as:

    Error: ENOENT: no such file or directory, open '/home/utku/mdn/content/files/files/_githistory.json'

when running:

    node tool/cli.js gather-git-history --verbose

peterbe commented:
> @peterbe I can't seem to build it locally. I give `content/files` as `CONTENT_ROOT` but the build fails as:
> 
> ```
> Error: ENOENT: no such file or directory, open '/home/utku/mdn/content/files/files/_githistory.json'
> ```
> 
> when running:
> 
> ```
> node tool/cli.js gather-git-history --verbose
> ```

Weird! 
I don't think you're the first one to run into this. 
First of all, it seems suspicious that `files` is mentioned twice in `/home/utku/mdn/content/files/files/_githistory.json`
Are you using any symbolic links?
What if you try to not say the relative path. What if you instead try something like this:
```
export CONTENT_ROOT=/home/utku/mdn/content/files
yarn dev
```

Also, why did you run `node tool/cli.js gather-git-history --verbose` in the first place?

ugultopu commented:
@peterbe I was running the build with `yarn dev`, not running `node tool/cli.js gather-git-history --verbose` by itself. Sorry for being ambiguous there.

OK, fixed it. Initially I set `CONTENT_ROOT` to `/path/to/content` instead of `/path/to/content/files` and ran `yarn dev` that way. After it finished, I realized that it was wrong and then I set `CONTENT_ROOT` to the correct path (which is `/path/to/content/files`). Then I started getting that error.

After some inspection, I realized that the build was using a file named `yari-git-history.json` in a subfolder of `/var/folders`. After I removed it, it worked. So not sure exactly how to reproduce it, but maybe the code over there can be improved to prevent this.

Later, I compared the full path of `yari-git-history.json` that the build was using and realized that the path is equivalent to:

```javascript
const { join } = require('path');
const { tmpdir } = require('os');

join(tmpdir(), 'yari-git-history.json`);
```

So, I think if the user has initially set a wrong path for `CONTENT_ROOT`, even after setting it to the correct path, this error will happen.

ugultopu commented:
@peterbe OK, I reproduced it. I'm going to create an issue so that it can be solved later.

schalkneethling commented:
@ugultopu Hey there and thank you so much for your contribution. My sincere apologies for only responding here now especially with what I am about to suggest. 

Two things:

1. I would like to discuss this particular issue a bit before we decide to merge a solution(https://github.com/mdn/mdn-minimalist/issues/440), and
2. The CSS for this should ideally be added to mdn/minimalist and not directly to the codebase here

I would therefore ask if you would be open to closing this pull request and creating a new pull request on https://github.com/mdn/mdn-minimalist/ against https://github.com/mdn/mdn-minimalist/issues/440 - Depending on the outcome of the discussion on the issue we can then further explore this solution from a code perspective.

Thanks again @ugultopu, and I hope this does not cause too much inconvenience for you. 🙏

ugultopu commented:
No problem. I think the decision is to replace them with notecards. I guess they already have styles so the only change needs to be done in content? Anything that fixes the styling so that is easier to read the articles is a good solution 😊 My intention was to create a possibly temporary solution so that the readers won't have a hard time until the real solution has been implemented.

schalkneethling commented:
> No problem. I think the decision is to replace them with notecards. I guess they already have styles so the only change needs to be done in content?

100% correct

ugultopu commented:
On another note, the content is still not changed yet. Please take this as a constructive criticism but if this was my project, I would approve this PR (or any solution that fixes the styling) and immediately create:

- One new PR in this repository that reverts the changes of this PR.
- One issue on the content repository named "Migrate all callout boxes to notecards" (let's say that this issue's number will be `N`)

On the new PR's comment, I would say "This PR can be merged only after issue `N` on the content repository has been successfully closed". 

This way, the readers would immediately start reading styled articles instead of still reading unstyled ones.
